qiskit_machine_learning/circuit/library/raw_feature_vector.py

```diff
@@ -112,11 +117,13 @@ def num_qubits(self, num_qubits: int) -> None:
         Args:
             The new number of qubits.
         """
-        if self._num_qubits != num_qubits:
+        if self.num_qubits != num_qubits:
             # invalidate the circuit
             self._invalidate()
-            self._num_qubits = num_qubits
-            self.add_register(QuantumRegister(self.num_qubits, "q"))
+            if num_qubits:
```

Comment:
`if <variable>` is true for many different inputs and that led to some difficult to find bugs in the past. I think it's better to explicitly check for the value 🙂 
```suggestion
            if num_qubits > 0:
```

Reply:
Good point 💫

Reply:
Yes that would be good. Do you want to open a PR to fix it? 😇

Reply:
Sure